Raise an error when calling make with MARCHes known to fail

nalimilan opened the pull request:
Pre-SSE2 CPUs are not supported as some tests fail (#7185).
Better warn as soon as possible.

yuyichao commented:
The only issue I can see is that some distro maintainer might want to support older architectures (and for cases it works reasonably well...).

nalimilan commented:
Yeah, this was discussed on the other issue. For Fedora, I'm supposed to make i386 or i686-compatible builds. But in practice, better a Pentium4 build than nothing, and I don't think anybody would notice.

Shipping a build which doesn't pass the tests sounds like a very bad idea to me, especially if that's just to pretend supporting machines that nobody uses.

yuyichao commented:
Good enough for me. (I've also changed my ArchLinux julia-git package to pentium4 and claiming `i686` arch)...

Just want to raise a concern I've seen from distro maintainers but if it has already been discussed I'll be totally find with it.

tkelman commented:
Actually we build 32-bit binaries right now with
```
MARCH = i686
JULIA_CPU_TARGET = pentium4
```
since there was a bug having to do with 32 bit calling convention in one of the dependencies, I think SuiteSparse.

nalimilan commented:
@tkelman Interesting. So `JULIA_CPU_TARGET` is what matters for this issue? I guess I should check for that variable instead, and suggest passing `MARCH=i686 JULIA_CPU_TARGET=pentium4` OR `MARCH=pentium4` if you don't build the dependencies. FWIW, the latter works fine in Fedora, and it would be a shame not to use SSE2 everywhere in Julia if we require it.

tkelman commented:
I agree but if we're going to adjust the buildbot configuration we should carefully test all 32-bit platforms on both release-0.3 and master first, to make sure it doesn't bring back any bugs.

nalimilan commented:
@tkelman Sorry, I had missed your comment. I wasn't really suggesting to change the buildbots right now -- just to give both suggestions to the user.

OK to merge the updated version?

tkelman commented:
I don't think I've built successfully with `MARCH=i386`, I typically use `i686`.

nalimilan committed:
Raise an error for JULIA_CPU_TARGET values known to fail

Pre-SSE2 CPUs are not supported as some tests fail (#7185).
Better warn before the build.

nalimilan commented:
OK, changed to `i686`.

tkelman commented:
I pushed a commit here to make sure this doesn't break 32 bit appveyor

tkelman commented:
my mistake, sorry, I already covered this in https://github.com/JuliaLang/julia/blob/f3a6461215ba2f8ce4a7ace7923503aa6f8f6ecb/contrib/windows/msys_build.sh#L49
